nss/integration-tests/nss_test.go

```diff
@@ -0,0 +1,168 @@
+		"Error when getting group by name and entry does not exist":  {db: "group", key: "doesnotexit", wantErr: true},
+		"Error when getting shadow by name and entry does not exist": {db: "shadow", key: "doesnotexit", wantErr: true},
+
+		"Error when getting passwd by id and entry does not exist": {db: "passwd", key: "404", wantErr: true},
+		"Error when getting group by id and entry does not exist":  {db: "group", key: "404", wantErr: true},
+	}
+	for name, tc := range tests {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			if tc.cacheDB == "" {
+				tc.cacheDB = "multiple_users_and_groups"
+			}
+
+			var socketPath string
+			if !tc.noDaemon && !tc.noCustomSocket {
+				ctx, cancel := context.WithCancel(context.Background())
+				socketPath = runDaemon(ctx, t, tc.cacheDB)
+				t.Cleanup(cancel)
+			}
+
+			cmds := []string{"getent", tc.db}
+			if tc.key != "" {
+				cmds = append(cmds, tc.key)
+			}
+
+			got, err := outNSSCommandForLib(t, socketPath, originOuts[tc.db], cmds...)
+			time.Sleep(time.Second)
+			if tc.wantErr {
+				require.Error(t, err, "Expected an error, but got none")
+				return
+			}
+			require.NoError(t, err, "Expected no error, but got %v", err)
+
+			want := testutils.LoadWithUpdateFromGolden(t, got)
+			require.Equal(t, want, got, "Outputs must match")
+		})
+	}
+}
+
+func TestMain(m *testing.M) {
+	testutils.InstallUpdateFlag()
+	flag.Parse()
+
+	execPath, cleanup, err := buildDaemon()
+	if err != nil {
+		log.Printf("Setup: failed to build daemon: %v", err)
+		os.Exit(1)
+	}
+	defer cleanup()
+	daemonPath = execPath
+
+	// Creates the directory to store the coverage files for the integration tests.
+	if testutils.WantCoverage() {
+		rawCovDir = os.Getenv("GOCOVERDIR")
+		if rawCovDir == "" {
+			dir, err := os.MkdirTemp("", "authd-coverage")
+			if err != nil {
+				log.Printf("Setup: failed to create temp dir for coverage: %v", err)
+				cleanup()
+				os.Exit(24)
+			}
+			defer os.RemoveAll(dir)
+			rawCovDir = dir
+		}
+	}
+
+	code := m.Run()
+	if code == 0 && rawCovDir != "" {
+		coverprofile := filepath.Join(rawCovDir, "integration-tests.coverprofile")
+		// #nosec:G204 - we control the command arguments in tests
+		cmd := exec.Command("go", "tool", "covdata", "textfmt", fmt.Sprintf("-i=%s", rawCovDir), fmt.Sprintf("-o=%s", coverprofile))
+		if err := cmd.Run(); err != nil {
+			log.Printf("Teardown: failed to parse coverage files: %v", err)
+			cleanup()
+			os.RemoveAll(rawCovDir)
+			os.Exit(24)
+		}
+		testutils.MarkCoverageForMerging(coverprofile)
+	}
+
+	if err := testutils.MergeCoverages(); err != nil {
+		log.Printf("Teardown: failed to merge coverage files: %v", err)
+
+		// This ensures that we fail the test if we can't merge the coverage files, if the test
+		// was successful, otherwise we exit with the code returned by m.Run()
+		if code == 0 {
+			cleanup()
+			os.RemoveAll(rawCovDir)
+			defer os.Exit(24)
```

Comment:
Honestly, it was just a random exit status (that doesn't match any of the returned statuses by `go test`) picked at random, so we could know that the tests failed for this specific reason. It could be changed for something else.